src/idc.py

```diff
@@ -0,0 +1,399 @@
+        operand is a register phrase   => phrase number
+        otherwise                      => -1
+    """
+    program_api, min_address = get_program_context()
+    listing = cp.currentProgram.getListing()
+    code_unit = listing.getCodeUnitAt(program_api.toAddr(ea + min_address))
+
+    if code_unit is None:
+        return -1
+
+    insn = ida_ua.insn_t()
+    inslen = ida_ua.decode_insn(insn, ea)
+    if inslen == 0 or n >= code_unit.getNumOperands():
+        return -1
+
+    op = insn.ops[n]
+    if not op:
+        return -1
+
+    value = -1
+
+    if op.type & OperandType.REGISTER:
+        value = op.reg  # Register number
+    elif op.type & OperandType.SCALAR and not (op.type & OperandType.ADDRESS):
+        value = op.value  # Immediate value or displacement
+    elif op.type & OperandType.ADDRESS:
+        if op.type & OperandType.DATA or op.type & OperandType.SCALAR:
+            value = op.addr  # Direct memory reference
+        elif op.type & OperandType.CODE:
+            value = -1  # Code reference, not handled here
+
+    return value
+
+
+def next_head(ea, maxea=cp.currentProgram.maxAddress.getOffset()):
+    return ida_bytes.next_head(ea, maxea)
+
+
+def prev_head(ea, minea=cp.currentProgram.minAddress.getOffset()):
+    return ida_bytes.prev_head(ea, maxea)
+
+
+def get_struc_id(struc):
+    if struc == "EFI_GUID":
+        return 1
+    elif struc == "EFI_BOOT_SERVICES":
+        return 2
+
+
+def SetType(ea, newtype):
+    """
+    Set type of function/variable
+
+    @param ea: the address of the object
+    @param newtype: the type string in C declaration form.
+                    Must contain the closing ';'
+                    If specified as an empty string, the
+                    item associated with 'ea' will be deleted.
+
+    @return: 1-ok, 0-failed.
+    """
+    try:
+        fpa, min_addr = get_program_context()
+        addr = fpa.toAddr(min_addr + ea)
+        listing = cp.currentProgram.getListing()
+
+        if newtype == "":
+            # Delete any defined data at that address
+            existing = listing.getDefinedDataAt(addr)
+            if existing:
+                listing.clearCodeUnits(addr, addr, False)
+            return 1
+
+        parser = DataTypeParser(fpa, None)
+        dt = parser.parse(newtype)
+
+        DataUtilities.createData(
+            cp.currentProgram,
+            addr,
+            dt,
+            USE_DEFAULT_LENGTH,
+            False,
+            DataUtilities.ClearDataMode.CLEAR_ALL_UNDEFINED_CONFLICT_DATA,
+        )
+        return 1
+
+    except Exception as e:
+        print(f"[SetType] Failed to apply type '{newtype}' at {ea:#X}: {e}")
+        return 0
+
+
+def get_name(ea, flags=0):
+    """
+    Get name at the specified address
+
+    @param ea: linear address
+    @param gtn_flags: how exactly the name should be retrieved.
+                      combination of GN_ bits
+    """
+    pass
+
+
+def set_name(ea, name, flags=0):
```

Comment:
Why I am seeing a redefinition of these functions? ida_name.py, ida_api, and here.

This is the most important finding of this review: revise the design, there's something very wrong with this.